mobilizon_reshare/storage/query/write.py

```diff
@@ -13,6 +13,33 @@
 from mobilizon_reshare.storage.query.read import events_without_publications
 
 
+async def upsert_publication(publication_report, event):
+
+    publisher = await get_publisher_by_name(
+        name=publication_report.publication.publisher.name
+    )
+    old_publication = await Publication.filter(
+        id=publication_report.publication.id
+    ).first()
+    if old_publication:
+        # I update the existing publication with the new report
```

Comment:
if I don't do this, the id of the publication will change and possibly introduce problems somewhere. I considered introducing a versioning of the publication but it was out of scope for this PR and possibly complicating the code unnecessarily.

Reply:
I'm all for keeping the code simple, I have doubts because this is the only place we write in the Publication table so if we can avoid edge cases it's easier to reason about the whole state update process.

I think adding a new Publication for each trial shouldn't break anything and we have an history of trials for an event, publisher pair. we could decide to stop retrying after 5 failures or maybe something smarter.

You could try removing this if block and if the test suite runs I think we should be ok, what do you think?

Reply:
If the tests don't run and it becomes complicated to revert this change we can just merge it and think about this later

Reply:
I would be for merging this as it is and then open a different PR for publication history

Reply:
I can do it these days I think